src/extensions/default/InlineColorEditor/unittests.js

```diff
@@ -473,34 +473,77 @@ define(function (require, exports, module) {
+                    expect(PreferencesManager.get("uppercaseColors")).toBe(true);
+                });
+                it("should convert an uppercase hex color to rgb when mode button clicked", function () {
+                    testConvert("#112233", "rgba", "RGB(17, 34, 51)");
+                });
+                it("should convert an uppercase hex color to hsl when mode button clicked", function () {
+                    testConvert("#112233", "hsla", "HSL(210, 50%, 13%)");
+                });
+                it("should convert an uppercase rgb color to hex when mode button clicked", function () {
+                    testConvert("RGB(15, 160, 21)", "hex", "#0FA015");
+                });
+                it("should convert an uppercase rgba color to hex (dropping alpha) when mode button clicked", function () {
+                    testConvert("RGBA(15, 160, 21, 0.5)", "hex", "#0FA015");
+                });
+                it("should convert an uppercase rgb color to hsl when mode button clicked", function () {
+                    testConvert("RGB(15, 160, 21)", "hsla", "HSL(122, 83%, 34%)");
+                });
+                it("should convert an uppercase rgba color to hsla when mode button clicked", function () {
+                    testConvert("RGBA(15, 160, 21, 0.3)", "hsla", "HSLA(122, 83%, 34%, 0.3)");
+                });
+                it("should convert an uppercase hsl color to hex when mode button clicked", function () {
+                    testConvert("HSL(152, 12%, 22%)", "hex", "#313F39");
+                });
+                it("should convert an uppercase hsla color to hex (dropping alpha) when mode button clicked", function () {
+                    testConvert("HSLA(152, 12%, 22%, 0.7)", "hex", "#313F39");
+                });
+                it("should convert an uppercase hsl color to rgb when mode button clicked", function () {
+                    testConvert("HSL(152, 12%, 22%)", "rgba", "RGB(49, 63, 57)");
+                });
+                it("should convert an uppercase hsla color to rgba when mode button clicked", function () {
+                    testConvert("HSLA(152, 12%, 22%, 0.7)", "rgba", "RGBA(49, 63, 57, 0.7)");
+                });
+                
+                // Use lowercase colors
+                PreferencesManager.set("uppercaseColors", false);
+                
+                it("should use lowercase colors", function () {
+                   expect(PreferencesManager.get("uppercaseColors")).toBe(false);
+                });
+                it("should convert an lowercase hex color to rgb when mode button clicked", function () {
                     testConvert("#112233", "rgba", "rgb(17, 34, 51)");
                 });
-                it("should convert a hex color to hsl when mode button clicked", function () {
+                it("should convert an lowercase hex color to hsl when mode button clicked", function () {
```

Comment:
It looks like all of these test are lowercase to lowercase or uppercase to uppercase -- I think there should be some tests for lowercase to uppercase and uppercase to lowercase, too.

Also, what is the expectation when original text is mixed case? I think there should be some tests added for that.